lib/rust/types/src/unit2.rs

```diff
@@ -0,0 +1,631 @@
+//! Utilities for creating custom strongly typed units. For example, unit `Duration` is a wrapper
```

Comment:
I’m wondering whether we would not have been better off to use a pre-existing solution than re-implementing this ourselves and now having two different implementations of this in our code base. 

I'm not going to argue against including this as it is already written, but I think it might have been easier to use something off the shelves (e.g. `uom` or `chrono`) for our use case, which would lead to a smaller codebase on our side, making future maintenance easier for us.
 We should also strongly consider planning a chore to remove the old one (or both) in the midterm.

Reply:
We have many units, including pixels, or some shader units. There is no package I know that provides. generic way of working with units. Also `unit2` is way better than `unit`, so I want to slowly migrate to it and remove the old `unit` over time.